### PR TITLE
Migrate e2e test for HashiCorp Vault to Go.

### LINKED ISSUE #3176: Migrate e2e test for Hashicorp Vault to Go.
### Proposal

Relates to https://github.com/kedacore/keda/issues/2357, https://github.com/kedacore/keda/issues/2737.

### Use-Case

_No response_

### Anything else?

_No response_

### PR DESCRIPTION
**Pull request by v-shenoy**
Signed-off-by: Vighnesh Shenoy <vshenoy@microsoft.com>

_Provide a description of what has been changed_

### Checklist

- [x] Commits are signed with Developer Certificate of Origin (DCO - [learn more](https://github.com/kedacore/keda/blob/main/CONTRIBUTING.md#developer-certificate-of-origin-signing-your-work))
- [x] Tests have been added

Fixes #3176 

Relates to #2737

### CODE REVIEW

**Comment by v-shenoy** [comment]
This is failing on my machine and as far as I can tell it's because the vault is sealed. This shouldn't be the case as I am passing `--set "server.dev.enabled=true"`. 

Feels like I am making some stupid error. @zroubalik

File: tests/secret-providers/hashicorp_vault/hashicorp_vault_test.go [hunk 1]
```diff
@@ -0,0 +1,382 @@
...
+func setupHashiCorpVault(t *testing.T, kc *kubernetes.Clientset) string {
+	CreateNamespace(t, kc, vaultNamespace)
+
+	_, err := ExecuteCommand("helm repo add hashicorp https://helm.releases.hashicorp.com")
+	assert.NoErrorf(t, err, "cannot add hashicorp repo - %s", err)
+
+	_, err = ExecuteCommand("helm repo update")
+	assert.NoErrorf(t, err, "cannot update repos - %s", err)
+
+	_, err = ExecuteCommand(fmt.Sprintf(`helm upgrade --install --set "server.dev.enabled=true" --namespace %s --wait vault hashicorp/vault`, vaultNamespace))
```
Comment (JorTurFer) [inline tests/secret-providers/hashicorp_vault/hashicorp_vault_test.go:304]: remove the quotes around `"server.dev.enabled=true"`. I think that using our approach, it makes something weird. At least I had this problem with datadog chart

Comment (JorTurFer) [reply tests/secret-providers/hashicorp_vault/hashicorp_vault_test.go:304]: In fact, if you check the chart values when you install it with quotes, you will see that the value is not parsed and it's registered as another different value with the quotes. At least, that was my case, all the values with quotes weren't parsed and was registered wrong

Comment (v-shenoy) [reply tests/secret-providers/hashicorp_vault/hashicorp_vault_test.go:304]: I thought so the problem was with the way it was being parsed, but I couldn't figure out how. I ran a [snippet](https://go.dev/play/p/7lFsBrrjOpZ) on the go playground, and it splits the way I expect it to. Not sure what the issue is. Maybe something in the internals of the `exec` package and the way it executes the command?

Comment (JorTurFer) [reply tests/secret-providers/hashicorp_vault/hashicorp_vault_test.go:304]: Maybe... Have you tried without the quotes? Does it work?

Comment (v-shenoy) [reply tests/secret-providers/hashicorp_vault/hashicorp_vault_test.go:304]: Yes, it does as I mentioned above. I am just not sure why :P

Comment (JorTurFer) [reply tests/secret-providers/hashicorp_vault/hashicorp_vault_test.go:304]: This is [the line](https://github.com/kedacore/keda/blob/main/tests/scalers_go/datadog/datadog_test.go#L303) I talked about

Comment (zroubalik) [reply tests/secret-providers/hashicorp_vault/hashicorp_vault_test.go:304]: I think that it doesn't like the command and all agrs together in one `string`, it should be probably `[]string` passed to the `exec`. I do recall having some problems around this somewhere in the past 😅

Comment (v-shenoy) [reply tests/secret-providers/hashicorp_vault/hashicorp_vault_test.go:304]: @zroubalik This is what is being done in the helper package. The `string` is split into `string[0]` and `string[1:]` and passed to `exec`.

Comment (zroubalik) [reply tests/secret-providers/hashicorp_vault/hashicorp_vault_test.go:304]: Yeah, the exec accepts `[]string`, so there's a high chance that it is causing the problems:
https://github.com/kedacore/keda/blob/17b78dd66f63ec58981facd1b78aad5114302456/tests/helper/helper.go#L113-L124

Comment (v-shenoy) [reply tests/secret-providers/hashicorp_vault/hashicorp_vault_test.go:304]: https://github.com/kedacore/keda/blob/17b78dd66f63ec58981facd1b78aad5114302456/tests/helper/helper.go#L64-L79

I'm referring to this.

Comment (zroubalik) [reply tests/secret-providers/hashicorp_vault/hashicorp_vault_test.go:304]: Ah right, I thought that this command is run on a pod.

Comment (v-shenoy) [reply tests/secret-providers/hashicorp_vault/hashicorp_vault_test.go:304]: https://go.dev/play/p/bBitQgGZ7qk

Running this snippet and printing the `exec.Cmd` object prints the expected command. I am so confused right now.

**Review by JorTurFer — APPROVED** [review]

**Pull request merged**